sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -100,9 +105,13 @@ def _generate_recommendations(self):
             self.line_ids += new_line
             existing_product_ids.append(line.product_id.id)
         # Add those recommendations too
+        products_allowed_ids = self.env['product.product'].browse(
+            [l["product_id"][0] for l in found_lines]).filtered(
+            lambda x: x.active and x.sale_ok).ids
```

Comment:
I'd say this makes more sense here:

```python
products_allowed_ids = set(self.env["product.product"].search([
    ("id", "in", [l["product_id"][0] for l in found_lines]),
    ("id", "not in", existing_product_ids),
    ("active", "=", True),
    ("sale_ok", "=", True),
).ids)
```

You can even remove the `active=true` test because it's automatic when you're filtering on the main model. I left it just to let you understand better the change.

If you're gonna make a query, then at least you save the latter python-side filtering.